### ISSUE TITLE
Document title when opening a page via Site Editor's Navigation panel is a bit confusing

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#48641]
A long time ago in a galaxy far far away there was a switching mechanic in the site editor Top Toolbar that allowed you to load arbitrary content into the selected template, it looked a bit like this:

<img width="1079" alt="Screenshot 2023-03-01 at 14 51 51" src="https://user-images.githubusercontent.com/846565/222175589-6a5dc893-3239-4917-a433-99a7edb30588.png">

Later this was removed in order to focus more on the template editing experience. 

However, with the addition of the Navigation panel, it once again becomes possible to load different items of content into a template, albeit via slightly different flow. There results an issue with the optics... it's a bit confusing when you go to edit your "Blog" page and find the subsequent title to be "Index":

https://user-images.githubusercontent.com/846565/222176917-74e196ce-00a9-4299-87b8-c805a3b21d90.mp4

It might make sense to restore the labelling from before, whereby users can interpret not only which template they're editing, but the content too.

**Comment by jameskoster** [comment]
Here's a first pass:

**Editing a template alone**

<img src="https://user-images.githubusercontent.com/846565/222203399-0581de23-97d1-410c-adb5-3cf98876ffad.png" width="500">

* `Layout` icon indicates you’re editing a template
* Template details popover appears on click

---

**Editing Page template with specific page loaded** 

<img src="https://user-images.githubusercontent.com/846565/222203392-d6068e86-c8e0-43ec-b986-22023cd6b935.png" width="500">

**Editing Single Post template with specific post loaded**

<img src="https://user-images.githubusercontent.com/846565/222203394-9c583df3-9633-4180-8e09-ea5b0b8fe8a2.png" width="500">

**Editing Archive template with specific tag loaded**

<img src="https://user-images.githubusercontent.com/846565/222203396-8bc839dd-a829-4349-997a-b5661ab09afd.png" width="500">

* Template chip accompanied by associated content chip
* Content icon helps clarify between posts, pages, category / tag archives, etc.
* Long titles get truncated

The template chip appears first because the primary tab in the Inspector will continue to be 'Template'.

I don't think the content chip needs to be interactive to begin with. But in the future it might serve as a tool to help illustrate whether the selected block is part of the template, part of the content, or both. Rough example:


https://user-images.githubusercontent.com/846565/222208963-3511d542-681c-4e5f-82e2-98217e899fe9.mp4



When the selected block does not affect a scope, the associated chip is de-emphasised.

**Comment by SaxonF** [comment]
> it's a bit confusing when you go to edit your "Blog" page and find the subsequent title to be "Index":

This is definitely the key pain point to resolve. With navigation in place, are we ready to just lean fully into being content first? In that world we wouldn't be showing what page/post a template has loaded but rather which template a post/page has applied. After clicking a page name in navigation I'd expect to see that listed first in toolbar. In future we may not even want to show template name in toolbar and introduce an "edit template" step to get there.

If it's a collection based page (like tag), the approach here works really well.

To summarise thoughts
- Introduce icon
- If editing a "page" (entry via navigation panel), show page name first. In future, if we do introduce an "edit template" action, consider showing just page name
- otherwise (entry via templates panel) show template name first

**Comment by jameskoster** [comment]
My feeling is that we can follow up to get the ordering right. I lean towards putting template first only because the Inspector will still indicate that you're editing the template:

<img width="313" alt="Screenshot 2023-03-02 at 10 43 06" src="https://user-images.githubusercontent.com/846565/222406094-525d174b-b97f-4acb-b643-ffd41a5132d1.png">

Of course we should 'fix' that too, but doing so likely needs a dedicated effort to figure out details like whether to display both tabs.

**Comment by jasmussen** [comment]
A few different strands coming together here:

* Content/template split as mainly mocked up here
* "Command bar" access

The latter I've seen a mockup that's a bit like this:

<img width="1756" alt="Search bar" src="https://user-images.githubusercontent.com/1204802/231453392-c28a07df-0999-4e72-9041-1a9874717b0f.png">

I quite like your mockups here, Jay, with the content/template split. On a separate sketch I believe from Saxon, he also introduced the purple template color, to good effect:

<img width="827" alt="Screenshot 2023-04-12 at 14 13 28" src="https://user-images.githubusercontent.com/1204802/231453816-01a1a73a-c804-48ce-990c-b1ebd9f1ac04.png">

The two approaches (command bar vs. breadcrumbs) seem at odds, though, at least in the current designs. Can they be reconciled? The gray round-rect could certainly become smaller, but which of the two items in the breadcrumb would be invoked on ⌘K?

**Comment by jameskoster** [comment]
> with the content/template split

I think there's a general question to be answered about how much emphasis the UI should place on template editing during a content editing flow.

Having played a bit with [this branch](https://github.com/WordPress/gutenberg/pull/49292) I'm beginning to think that the top bar placement might be a bit too prominent, so I lean more towards the 'url bar' approach (containing a single label).

For the label, perhaps that should be linked with the out-of-view details panel – IE whatever is loaded into that panel is used for the label in the url bar. When it's a template, we can potentially use the purple colorisation.

**Comment by Mamaduka** [comment]
Hey folks 👋

I'm going to punt this enhancement to WP 6.3. The enhancements are rarely shipped with minor versions.

**Comment by mtias** [comment]
Here's a list of things the doc title could allow a user to do:

- Quick jump to other pages or templates or recently visited ones (command center).
- Rename the page or change the URL.
- Could absorb a "view" link.
- Template / Content association.

Doesn't have to do all of these, but would be good to narrow down on the function.

**Comment by jasmussen** [comment]
To connect some dots, I created a mockup in https://github.com/WordPress/gutenberg/issues/50378#issuecomment-1538173078 that overlaps a bit with this one. Should we consolidate?

**Comment by jameskoster** [comment]
Yup closing #50378 will fix this.
